File: lista_1/test_iterative_lib.c
#include <stdio.h>
#include "iterative_lib.h"

int main() {
    printf("Biblioteka iteracyjna \n\n");
    unsigned int n;
    printf("Podaj liczbe naturalna do obliczenia silni: ");
    scanf("%d", &n);
    printf("%d! = %lld\n", n, factorial(n));

    unsigned int a, b;
    printf("\nPodaj dwie liczby naturalne do obliczenia NWD: ");
    scanf("%u %u", &a, &b);
    printf("NWD(%u, %u) = %u\n", a, b, gcd(a, b));

    int x, y, c;
    printf("\nPodaj wspolczynniki a, b i c dla rownania diofantycznego ax + by = c: ");
    scanf("%d %d %d", &a, &b, &c);
    Solution sol = diophantine(a, b, c);
    printf("Error: %s\n", sol.err ? "true" : "false");
    printf("Rozwiazanie: x = %d, y = %d\n", sol.x, sol.y);

    return 0;
}

File: lista_1/python_library/test_iterative_lib.py
# test_iterative_lib.py
from iterative_lib import factorial_iterative, gcd_iterative, diophantine_iterative

print("Biblioteka iteracyjna\n")

n = int(input("Podaj liczbę naturalną do obliczenia silni: "))
print(f"{n}! = {factorial_iterative(n)}")

a, b = map(int, input("\nPodaj dwie liczby naturalne do obliczenia NWD: ").split())
print(f"NWD({a}, {b}) = {gcd_iterative(a, b)}")

a, b, c = map(int, input("\nPodaj współczynniki a, b i c dla równania diofantycznego ax + by = c: ").split())
sol = diophantine_iterative(a, b, c)
print(f"Error: {sol.err}")
print(f"Rozwiązanie: x = {sol.x}, y = {sol.y}")
